### PR TITLE
Define `Group.__getitem__()` method

### PR DESCRIPTION
**Pull request by ankostis**
Since `tables.group.Group` class defines `__contains__()` and `iter__()` methods, it should also define `__getitem__()`, and work as a proper *container*.  

That would make the code below to work intuitively:

```python
>>> 'group 1' in hfile.root
>>> g1 = hfile.root['group 1']
>>> g1['some &^%&^* named group']._v_pathname 
/group 1/some &^%&^* named group
```
For the record, that must have been the purpose of #270.

### Bonus commit: 
Improv unique checks on `__dir__()` TCs.

### CODE REVIEW

**Comment by tomkooij** [comment]
Thanks for the extra testcases.

About `Group.__getitem__`:

` >>>'group 1' in hfile.root`
This already works.

`>>> g1 = hfile.root['group 1']`
This is very ugly. 
The syntax `g1 = hfile.get_node('/group1')` is much more clean IMHO.

I won't **block** `__getitem__` but I do not feel that is it a necessary extension. I'd like to see what other devs think. pinging @PyTables/core

**Comment by ankostis** [comment]
In python it is very much expected if `foo in group` works, also `group[foo]` to work;  it is the read-only ["container" data-model](https://docs.python.org/3/reference/datamodel.html#emulating-container-types).  What is more puzzling, is that `atttribute_sets` work already like containers, but `groups` not.

> ```>>> g1 = hfile.root['group 1']```
> This is very ugly.

It's strange that you say so :-)
I would say, having everything working as a containers is the "zen" of python, but that's my opinion...

Anyway, I will implement it, and you may decide about it.
Personally I'm always monkey-patching `Group` class, not to have to remember in which cases indexing works or not.

**Comment by ankostis** [comment]
I forgot to explain that I referring to interactive use, e.g. Jupter notebooks, where you use the `[Tab]` completion to review the existing children, and then type as few as possible keystrokes to access the child. And work your way gradually doen the hierarchy.
I had this moment today where I was presenting my notebook to a colleague and had to explain why simple indexing did not function, and dig my way through `_v_...` methods.

**Comment by FrancescAlted** [comment]
Yeah, I tend to agree that PyTables philosophy is to treat children as attributes (i.e. using 'natural naming'), and not as entries in a dictionary.  Having said that, practicality beats purity, and if people like using the `Group` class as a dictionary, then I don't see strong reasons so as to not allowing that. So +1 on accepting this PR.

@ankostis I don't think I get your point in what the trouble is using `[TAB]` in the Jupyter notebook.  This should work in PyTables, or it does not for you?

**Comment by ankostis** [comment]
Well, I was a bit confusing in my previous posting.
I had irregular-named groups, so `[Tab]` would not reveal them, but console was printing them, so I was trying expressions like `root['some child'].other_child.more['(g2)']` and they didn't work, and had a hard time finding which method to use to achieve this "gradual descend".  

I fixed my posting above.

File: RELEASE_NOTES.txt [hunk 1]
```diff
@@ -24,6 +24,9 @@ Improvements
 ------------
  - setup.py detects conda env and uses installed conda (hdf5, bzip2, lzo
    and/or blosc) packages when building from source.
+ - On interactive python sessions, group/attribute  `__dir__()` method
```
Comment (tomkooij) [inline RELEASE_NOTES.txt]: This was not released yet. `v3.4.2` does not include PR 624 and PR 625. Please move this to "Changes from 3.4.2 to 3.4.x".

**Comment by ankostis** [comment]
Release-changes fixed as requested.

**Comment by tomkooij** [comment]
Thanks!

File: tables/group.py [hunk 2]
```diff
@@ -829,6 +839,9 @@ def __getattr__(self, name):
             return self.__dict__[name]
         return self._f_get_child(name)
 
+    def __item__(self, name):
+        return self._f_get_child(name)
+
```
Comment (avalentino) [inline tables/group.py]: Sorry, probably I'm missing something, but what is for  `__item__` exactly? If I'm not it is not a standard dunder.

Comment (tomkooij) [inline tables/group.py]: O... I missed that in my review... 

I'll remove it.

Comment (tomkooij) [inline tables/group.py]: Removed in cc1175425397b4906ef909538345d4daa815029a

Comment (avalentino) [inline tables/group.py]: thanks

**Pull request merged**